rir/src/compiler/analysis/loop_detection.h

```diff
@@ -63,6 +66,35 @@ class LoopDetection {
             o->isInnermost_ = false;
         }
 
+        /*
+         * TODO: finds a preheader when there is one.
+         * We should create a preheader when there is non.
+         */
+        BB* preheader(const CFG& cfg) {
```

Comment:
Yeah, I'm looking at where you construct a `LoopDetection` class, and there you have to compute the CFG again... which `LoopDetection`'s constructor also has to do.

Reply:
Yes, recomputing the CFG is a little ugly I realized it at the moment and agree.
But, I still think we should compute the preheader lazily. Otherwise, our loop will depend on the fact that the CFG up to its header remains immutable which I see as a pretty strong constraint.